### PR TITLE
don't insert Last-Modified: if exits. (#632)

### PR DESCRIPTION
**Pull request by kazu-yamamoto**
@Profpatsch Would you review this?
Cc: @nh2

### CODE REVIEW

File: warp/Network/Wai/Handler/Warp/File.hs [hunk 1]
```diff
@@ -34,12 +34,17 @@ data RspFileInfo = WithoutBody H.Status
 ----------------------------------------------------------------
 
 conditionalRequest :: I.FileInfo
-                   -> H.ResponseHeaders -> IndexedHeader
+                   -> H.ResponseHeaders
+                   -> IndexedHeader -- Response
+                   -> IndexedHeader -- Request
```
Comment (nh2) [inline warp/Network/Wai/Handler/Warp/File.hs]: `-- ^ Request` haddock comments might be better.

File: warp/Network/Wai/Handler/Warp/File.hs [hunk 2]
```diff
@@ -34,12 +34,17 @@ data RspFileInfo = WithoutBody H.Status
...
-conditionalRequest finfo hs0 reqidx = case condition of
+conditionalRequest finfo hs0 rspidx reqidx = case condition of
     nobody@(WithoutBody _) -> nobody
-    WithBody s _ off len   -> let !hs = (H.hLastModified,date) :
-                                        addContentHeaders hs0 off len size
+    WithBody s _ off len   -> let !hs1 = addContentHeaders hs0 off len size
+                                  !hasLM = isJust $ rspidx ! fromEnum ResLastModified
+                                  !hs
+                                    | hasLM     = hs1
+                                    | otherwise = (H.hLastModified,date) : hs1
```
Comment (nh2) [inline warp/Network/Wai/Handler/Warp/File.hs]:
```haskell
!hs = [ (H.hLastModified,date) | not hasLM ] ++ hs1
```

might be considered easier to read by some (but that's controversial).

**Comment by Profpatsch** [comment]
Haven’t run it, but looks good to me. Might be worth adding a test for if there is a test suite.

**Comment by kazu-yamamoto** [comment]
@Profpatsch This patch includes a test case.

File: warp/Network/Wai/Handler/Warp/File.hs (hunk 1 above)
Comment (kazu-yamamoto) [reply warp/Network/Wai/Handler/Warp/File.hs]: Done

File: warp/Network/Wai/Handler/Warp/File.hs (hunk 2 above)
Comment (kazu-yamamoto) [reply warp/Network/Wai/Handler/Warp/File.hs]: Thanks and done.

**Comment by kazu-yamamoto** [comment]
Merged. Thank you all!

**Comment by nh2** [comment]
:tada:

**Comment by kazu-yamamoto** [comment]
A new version has been released.

**Pull request merged**